be/src/vec/exec/format/parquet/parquet_pred_cmp.h

```diff
@@ -240,6 +240,13 @@ class ParquetPredicate {
                               std::is_same_v<CppType, DateV2Value<DateV2ValueType>>) {
                     min_value.from_unixtime(min_date_value * 24 * 60 * 60, ctz);
                     max_value.from_unixtime(max_date_value * 24 * 60 * 60, ctz);
+
+                    // as VecDateTimeValue can not compare date and datetime, so need
+                    // cast to date here
+                    if constexpr (std::is_same_v<CppType, VecDateTimeValue>) {
```

Comment:
1. `date` is deprecated, and FE will always generated `datev2` currently, so how does `VecDateTimeValue `(for `date`) reach here?
2. `VecDateTimeValue ` has overload all compare operators, and doesn't distinguish `TIME_TIME` and `TIME_DATE` in `VecDateTimeValue`?
3. Even if the data type is `VecDateTimeValue `, how can there be a comparison between `date`  type and `datetime` type?